tests/weather-api.spec.js

```diff
@@ -1,14 +1,38 @@
 const should = require('should');
+const mongoose = require('mongoose');
 const weatherApi = require('../models/weather-api.js');
 const FORECAST_COUNT_PER_DAY = 8;
+const KEYS_OF_OBJECT_IN_WEATHER_ARRAY = ["hour", "day", "temp", "sky", "pty", "pop"];
+
 
 describe("Weather fetcher", function() {
-  it('promise return value should exist', function() {
-    return weatherApi.fetch().then(function(apiResponse){
-      should.exist(apiResponse);
-      // TODO: weather JSON에서 추출하는 로직을 따로 뺄 것.
-      var weather = apiResponse["rss"]["channel"]["item"]["description"]["body"]["data"];      
-      weather.length.should.be.aboveOrEqual(FORECAST_COUNT_PER_DAY);
-    });
+  var weatherArray = [];
+
+  before(function() {
+    return weatherApi.fetch().then(function(data) {
+      weatherArray = data;
+    })
+  });
+
+  it('promise return value should exist', function(done) {
+    should.exist(weatherArray);
+    weatherArray.length.should.be.aboveOrEqual(FORECAST_COUNT_PER_DAY);
+    done();
   });
+
+  it('Every element in Array has remanded keys and values', function(done) {
+    weatherArray.forEach(function(weatherElement) {
+      KEYS_OF_OBJECT_IN_WEATHER_ARRAY.forEach(function(key) {
+        weatherElement.should.have.properties(key);  
+      }, this);
+    }, this);
+    done();
+  });
+
+  it('Weather data filtered and parsed saved in mongoose', function() {
+    return weatherApi.addWeather(weatherArray).then(function(result){
+      should.exist(result);
+      result.should.have.properties("tomorrow");
+    });
+  })
```

Comment:
Suggestions:
- These tests should be changed since weather api module changed.
- **weatherApi.fetch()** is a core feature, so it should be in **it** block.